zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/CamundaOpenApiController.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.camunda.zeebe.gateway.rest.RestErrorMapper;
+import jakarta.annotation.PostConstruct;
+import java.nio.charset.StandardCharsets;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@CamundaRestController
+public class CamundaOpenApiController {
+
+  @Autowired private ObjectMapper objectMapper;
+
+  private String cachedJsonContent;
+
+  @PostConstruct
+  public void init() {
+    try {
+      final ClassPathResource resource = new ClassPathResource("apidoc/rest-api.yaml");
+      final String yamlContent =
+          new String(resource.getInputStream().readAllBytes(), StandardCharsets.UTF_8);
+
+      final var yamlMapper = new Jackson2ObjectMapperBuilder().factory(new YAMLFactory()).build();
+
+      final Object yamlObject = yamlMapper.readValue(yamlContent, Object.class);
+
+      cachedJsonContent = objectMapper.writeValueAsString(yamlObject);
+
+    } catch (final Exception e) {
+      throw new RuntimeException("Failed to load and parse OpenAPI YAML file at startup", e);
+    }
+  }
```

Comment:
❓ This can still happen multiple times now since this bean can be created multiple times in the application lifecycle, right? Not saying that's a blocker but wanted to raise this for understanding.

Reply:
To correct myself: I think this only happens once. Controller beans seem to be instantiated as singletons by Spring.

Reply:
You’re right in your follow-up: by default, Spring instantiates controllers as singletons, so the @PostConstruct method is called only once during the application lifecycle, right after the bean is initialized.

Reply:
I was wondering if we're trying to replicate something here though that Spring maybe already provides (serving static content that needs to be transformed before serving it). Could this be an alternative to do what we do here: https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-config/static-resources.html?

Reply:
I see what you mean. Yes, we would have to add a dedicated transformer to the resolver chain. I am not sure if we define static resource handling anywhere in the application already, I think we don't yet.

Ultimately, this here works, as far as I can tell. Having the post construct in a REST controller to load a resource so we can transform it, though, looks a little bit off to me. It simply looks like something Spring already solved with static resource controllers that can take transformers 😅

Reply:
No problem at all :) like I said, I was open to revisit

I updated the solution over this commit: https://github.com/camunda/camunda/pull/29628#:~:text=static%20resource%20handler-,495d5b0,-This%20branch%20has  (when we merge it, I am going to do the same for main)